### DISCUSSION TITLE
kong-revomatico oidc plugin is not able to connect keycloak as IDP

### DISCUSSION THREAD

**Discussion by shusriva** [revomatico/docker-kong-oidc#29] [Support]
Hi Team,

Kong revomatico oidc plugin is not able to resolve the discovery url for keycloak. 
I have deployed kong revomatico oidc plugin and keycloak in kubernetes cluster.

configuration.

apiVersion: configuration.konghq.com/v1
kind: KongPlugin
metadata:
  name: oidc
  #namespace: core
config:
  client_id: ${oidc_client_id}
  client_secret: ${oidc_client_secret}
  scope: openid
  realm: kong
  discovery: http://keycloak-discovery.core.svc.cluster.local:8080/auth/realms/master/.well-known/openid-configuration
plugin: oidc

issue:

accessing discovery url (http://keycloak-discovery.core.svc.cluster.local:8080/auth/realms/SCORE_DEV/protocol/openid-connect/auth) failed: [cosocket] DNS resolution failed: dns server error: 3 name error.

**Comment by cristichiru** [comment]
Well, there is your answer is in the failed DNS resolution.

Check that from your kong container you can curl/ping `keycloak-discovery.core.svc.cluster.local`. Of course, I assume there is a service called `keycloak-discovery` in the namespace `core` that you can access from the cluster nodes, or with an `externalIP` from outside.

Does any other service get resolved? Do you have a generic DNS issue within the cluster or is it just a typo with your keycloak service name?

We use the [official kong helm chart](https://github.com/Kong/charts/tree/main/charts/kong) to deploy kong that uses the default Kubernetes DNS policies: and works fine.
